### ISSUE TITLE
Atom feed should output both <summary> and <content>

### ISSUE THREAD

**Issue by eevee** [getpelican/pelican#1886]
The Atom feed seems to only spit out a `<summary>` tag.  Currently I have two options: populate it with summaries (which means the feed won't contain the full articles in any form, which is really annoying) or populate it with full articles (which is arguably against spec, and also means feed readers can't show a brief summary, which is also really annoying).

Unfortunately, from a brief glance through feedgenerator, it doesn't seem to support the `<content>` tag at all (!).

**Comment by justinmayer** [comment]
Hey Eevee. I believe FeedGenerator's current default behavior is indeed to use the `<summary>` tag to display whatever content/summary is fed to it, presumably because doing so is easier than determining/specifying whether the incoming content is a summary or the full content. I agree that a more elegant solution would be preferable. Is this something you would like to work on?

**Comment by eevee** [comment]
I could fix this, yeah, as long as feedgenerator is still accepting pull requests.  It hasn't had any activity (except a license change) in almost three years.

**Comment by justinmayer** [comment]
If you look at the [repository under the Pelican organization](https://github.com/getpelican/feedgenerator/commits/master), you'll see that there is more recent activity. Rest assured that we're still accepting pull requests.  (^_^)

**Comment by eevee** [comment]
Oh, excellent.  There sure are a lot of forks of that project floating around.  :)

**Comment by eevee** [comment]
So the required argument for feedgenerator's `add_item` is `description`, which sure sounds like it means a summary to me.  The [RSS spec](http://www.rssboard.org/rss-specification#hrelementsOfLtitemgt) would seem to agree, defining it as "The item synopsis."  Thus in my pull request (getpelican/feedgenerator#3), I kept the `description` as the Atom `<summary>`, and added a new optional `content` kwarg that becomes `<content>`.

But Pelican currently uses the entire post content as the `description`, and there's even a plugin devoted specifically to changing this to use the summary instead.  Arguably Pelican is doing the wrong thing here.  I'd like to change it to use the summary for `description` and the content for `content`...  which means RSS feeds will no longer contain the full content, because RSS has no field for that.  (Atom would have both, though feeds might look slightly different depending on feed reader.)

Is this reasonable, or should I keep the existing behavior?

**Comment by TierraDelFuego** [comment]
I tested this patch with my content and it only used the :summary: attributes from articles/pages. I have not tested the case of FEED_USE_SUMMARY = True and no articles/pages with :summary: attributes.

Also need to set FEED_USE_SUMMARY = True to pelicanconf.py.


```python
diff --git a/pelican/writers.py b/pelican/writers.py
index 423fc9f..adc4b96 100644
--- a/pelican/writers.py
+++ b/pelican/writers.py
@@ -45,13 +45,20 @@ class Writer(object):

         title = Markup(item.title).striptags()
         link = '%s/%s' % (self.site_url, item.url)
+        description = item.get_content(self.site_url)
+        try:
+            if self.settings['FEED_USE_SUMMARY']:
+                description = item.summary
+        except:
+            pass
+
         feed.add_item(
             title=title,
             link=link,
             unique_id='tag:%s,%s:%s' % (urlparse(link).netloc,
                                         item.date.date(),
                                         urlparse(link).path.lstrip('/')),
-            description=item.get_content(self.site_url),
+            description=description,
             categories=item.tags if hasattr(item, 'tags') else None,
             author_name=getattr(item, 'author', ''),
             pubdate=set_date_tzinfo(

```

**Comment by MyrionPhoenixmoon** [comment]
I just worked that into my local copy of Pelican, as I need the <content> tag to be populated.

It certainly works for me, so thank you!

**Comment by justinmayer** [comment]
Addressed via eac8fa2abd91823f3ffbdc679147504ee9489618.
